### PR TITLE
Fixes to remove old assignment and 10 char limit

### PR DESCRIPTION
**Pull request by Brunoga-MS**
# Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. Cleanup script [BUG 30923 - Several identities not found are left in the IAM of the MG](https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_backlogs/backlog/Azure%20Landing%20Zones%20Team/Stories/?workitem=30923)
2. Description and metadata addition to DINE-* policy files to correctly allow cleanup (https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_backlogs/backlog/Azure%20Landing%20Zones%20Team/Stories/?workitem=30923)
3. Added known issue about subsequent deployment to different locations
4. Removing 10 chars limitation for enterpriseScaleCompanyPrefix [BUG 30883 - Deployment on pseudo MG name longer than 10 chars fails](https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_backlogs/backlog/Azure%20Landing%20Zones%20Team/Stories/?workitem=30883)

### Breaking Changes

1. NONE

## As part of this Pull Request I have

- [ ] Read the Contribution Guide and ensured this PR is compliant with the guide
- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/azure-monitor-baseline-alerts/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/azure-monitor-baseline-alerts/issues) or ADO Work Items (Internal Only)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/azure-monitor-baseline-alerts/)
- [ ] Ensured PR tests are passing
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: patterns/alz/scripts/Start-AMBACleanup.ps1 [hunk 1]
```diff
@@ -135,7 +135,7 @@ ForEach ($identity in $policyAssignmentIdentities) {
 
     ForEach ($roleAssignment in $identityRoleAssignments) {
 
-        If ($roleAssignment.Description -like '*_deployed_by_amba*') {
+        If ($roleAssignment.Description -eq '_deployed_by_amba') {
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Start-AMBACleanup.ps1:138]: I see "_deployed_by_amba" was added in the metadata of the role assignment, however here the if statement the evaluations is done on ".Description" can you validate whether that is correct?

**Review by arjenhuitema — COMMENTED** [review]
Looks good, thank you Bruno. Just one comment about the metadata/description.  

Also, can you provide some evidence with these changes. For example, screenshot of a successful deployment, successful cleanup, deployment to a management group that has more than 10 characters, etc. Thanks!

File: patterns/alz/scripts/Start-AMBACleanup.ps1 (hunk 1 above)
Comment (Brunoga-MS) [reply patterns/alz/scripts/Start-AMBACleanup.ps1:138]: As far as the script line goes, I made the change to look for exact value (case insensitive). Wildcard in this case can lead to something different added by the customer ... very unlikely but possible.

Here they are the screenshots:
- Successful deployment with a pseudoMG name which is more than 10 chars 
image

- Description for role assignment correctly populated
image

- Successful cleanup
image

image

image

- Successful deployment
image

image

**Review by arjenhuitema — APPROVED** [review]
LGTM

**Pull request merged**